checker/pass_context_first_checker.go

```diff
@@ -0,0 +1,67 @@
+package checker
+
+import (
+	"fmt"
+	"go/ast"
+)
+
+func init() {
+	must(Register("pass_context_first", NewPassContextFirstChecker))
+}
+
+// PassContextFirstChecker checks that if a function declaration contains a context, it is
+// the first argument.
+type PassContextFirstChecker struct{}
+
+// NewPassContextFirstChecker constructs a PassContextFirstChecker.
+func NewPassContextFirstChecker(configData interface{}) NodeChecker {
+	return &PassContextFirstChecker{}
+}
+
+// Register implements the NodeChecker interface.
+func (c *PassContextFirstChecker) Register(fc *FileChecker) {
+	fc.On(&ast.FuncDecl{}, c)
+}
+
+// Check implements the NodeChecker interface.
+func (c *PassContextFirstChecker) Check(
+	node ast.Node,
+	content string,
+	report *Report) {
+
+	decl := node.(*ast.FuncDecl)
+
+	if decl.Type.Params == nil {
+		return
+	}
+
+	contextNotFirst := false
+	params := decl.Type.Params.List
+	if len(params) == 0 {
+		return
+	}
+
+	for _, param := range params[1:] {
+		selector, ok := param.Type.(*ast.SelectorExpr)
+		if !ok {
+			continue
+		}
+
+		ident, ok := selector.X.(*ast.Ident)
+		if !ok {
+			continue
+		}
+
+		if ident.Name == "context" && selector.Sel.Name == "Context" {
```

Comment:
You also need to handle the case where the `context` package is aliased in the scope of the file. Example:

```go
package x

import foo "context"

func bar(a int, ctx foo.Context) {
}
```

Reply:
Better add a TODO since this would require significant refactoring.

Reply:
Done.